public/index.html

```diff
@@ -10,40 +10,64 @@
   <div id="app">
     <button>Get your location geohash</button>
     <p></p>
+    <ul></ul>
   </div>
   <script src="/socket.io/socket.io.js"></script>
   <script>
     const text = document.querySelector("p");
     const btn = document.querySelector("button");
+    const ul = document.querySelector('ul');
 
-    let userHash = localStorage.getItem("userHash");
+    let user = JSON.parse(localStorage.getItem("user") || "{}");
 
-    if (userHash) {
-      text.innerHTML = `Your geohash is ${userHash}`;
+    if (user?.hash) {
+      text.innerHTML = `Your geohash is ${user.hash}`;
       btn.disabled = true;
       btn.innerText = "You are already signed in";
     }
 
+    const displayUserInfo =
+      (hash, username) => {
+        if (hash && username) {
+          text.innerHTML = `Your geohash is ${hash} and your username is ${username}`;
+        }
+      }
+
+    displayUserInfo(user?.hash, user?.username)
+
+
     const socket = io({
       query: {
-        hash: userHash,
+        hash: user?.hash,
+        username: user?.username,
```

Comment:
This change was made because the hash doesn't help us identify this user or other users to identify it. So, the user now locally stores their unique username.